### ISSUE TITLE
CLI updates for Infracost Cloud with Atlantis, Jenkins and other deployment systems

### ISSUE THREAD

**Issue by alikhajeh1** [infracost/infracost#1976]
1. The Infracost CLI needs to support the following env variables as overrides so Atlantis (and other deployment systems) can provide them manually. We already have `INFRACOST_VCS_REPOSITORY_URL`, `INFRACOST_VCS_SUB_PATH`, `INFRACOST_VCS_PULL_REQUEST_URL` and `INFRACOST_TERRAFORM_WORKSPACE`. Once the following is done, all top-level [run metadata](https://www.infracost.io/docs/features/cli_commands/#examples) can be overridden with env vars.
    ```
    === If a user has pull requests

    Required:
    INFRACOST_VCS_PROVIDER # github, gitlab, bitbucket, azure_repos
    INFRACOST_VCS_REPOSITORY_URL
    INFRACOST_VCS_PULL_REQUEST_URL
    INFRACOST_VCS_PULL_REQUEST_AUTHOR
    INFRACOST_VCS_PULL_REQUEST_TITLE

    Optional:
    INFRACOST_VCS_BASE_BRANCH
    INFRACOST_VCS_BRANCH
    INFRACOST_VCS_PIPELINE_RUN_ID
    INFRACOST_VCS_COMMIT_SHA
    INFRACOST_VCS_COMMIT_MESSAGE
    INFRACOST_VCS_COMMIT_TIMESTAMP
    INFRACOST_VCS_COMMIT_AUTHOR_EMAIL
    INFRACOST_VCS_COMMIT_AUTHOR_NAME
    INFRACOST_VCS_SUB_PATH
    INFRACOST_TERRAFORM_WORKSPACE

    === If the user does not have pull requests

    Required:
    INFRACOST_VCS_PROVIDER # github, gitlab, bitbucket, azure_repos
    INFRACOST_VCS_REPOSITORY_URL
    INFRACOST_VCS_BRANCH
    INFRACOST_VCS_COMMIT_SHA # added _SHA
    INFRACOST_VCS_COMMIT_MESSAGE
    INFRACOST_VCS_COMMIT_TIMESTAMP
    INFRACOST_VCS_COMMIT_AUTHOR_EMAIL
    INFRACOST_VCS_COMMIT_AUTHOR_NAME

    Optional:
    INFRACOST_VCS_PIPELINE_RUN_ID
    INFRACOST_VCS_SUB_PATH
    INFRACOST_TERRAFORM_WORKSPACE
    ```

2. Rename the Infracost JSON metadata `vcsRepoUrl` to `vcsRepositoryUrl`. That's the only one that's not consistently named with its env equivalent, `INFRACOST_VCS_REPOSITORY_URL`.

3. The Infracost CLI needs to be updated to set Atlantis and CircleCI vcs provider to the actual vcs provider, not "atlantis" or “circleci”. This'll enable Infracost Cloud to render icons for them correctly.

4. We should update the [3 examples in Atlantis](https://github.com/infracost/infracost-atlantis#2-how-do-you-want-multiple-terraform-directoriesworkspaces-to-be-handled) to set the required env vars for Infracost Cloud's new dashboard. Ideally we should do this for GitHub, GitLab, Azure Repos and Bitbucket e.g. for GitHub we need something like this:
    ```
    INFRACOST_VCS_PROVIDER=github # For GitHub Enterprise, also use "github"
    INFRACOST_VCS_PULL_REQUEST_URL=https://github.com/$BASE_REPO_OWNER/$BASE_REPO_NAME/pulls/$PULL_NUM
    INFRACOST_VCS_PULL_REQUEST_AUTHOR=$PULL_AUTHOR

    INFRACOST_VCS_PULL_REQUEST_TITLE=$(curl -s
      -H "Accept: application/vnd.github+json" \ 
      -H "Authorization: $GITHUB_TOKEN" \
      "https://api.github.com/repos/$BASE_REPO_OWNER/$BASE_REPO_NAME/pulls/$PULL_NUM" | jq -r '.title')
    ```

5. We should update the [Jenkins example](https://github.com/infracost/infracost-jenkins) to show how the above envs can be set for Infracost Cloud's new dashboard.

Once the above is done, we need to add an explicit way for users to mark the run as complete, or upload the results to Infracost Cloud, then the Infracost Cloud dashboard should work fine.

**Comment by alikhajeh1** [comment]
@hugorut I updated this issue to add `VCS_` to branch and commit env vars. Please update the issue desc if I got any wrong as I'll use this issue to write the docs for these envs.

**Comment by hugorut** [comment]
@alikhajeh1 I've actually decided to keep the old non `VCS_` branch vars as I think the separation you had before was useful. Let me know if you think there's a strong reason to change to the `VCS_` naming for those.

**Comment by hugorut** [comment]
Additional env var support is now merged in master and will be available with the next release. I'm keeping this issue open as it still depends on https://github.com/infracost/infracost/pull/1997 (`infracost upload` support)

**Comment by hugorut** [comment]
This functionality has been published in the [0.10.11](https://github.com/infracost/infracost/releases/tag/v0.10.11) release.
